Realm/RLMResults.mm

```diff
@@ -364,6 +364,41 @@ - (RLMResults *)sortedResultsUsingDescriptors:(NSArray<RLMSortDescriptor *> *)pr
     });
 }
 
+- (RLMResults *)distinctResultsUsingKeyPaths:(NSArray<NSString *> *)keyPaths {
+    if (keyPaths.count == 0) {
+        @throw RLMException(@"Must specify at least one keypath.");
+    }
+    for (NSString *keyPath in keyPaths) {
+        if (keyPath == nil
+            || [[keyPath stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceAndNewlineCharacterSet]]
```

Comment:
I'm not quite sure what this is doing. Whitespace isn't valid in key paths.

Reply:
Right, that's why the check is there. If someone passes a nonsense empty key path for some reason, it will throw early. If you don't feel it's necessary, I'm happy to remove it though.

Reply:
Key paths like that will already be rejected when attempting to look their components up as property names. It doesn't strike me as useful to add extra checking for the unusual case of passing an empty or white-space only string. There are plenty of other invalid key paths that we're not explicitly checking for here.